src/libstore/content-address.hh

```diff
@@ -5,44 +5,75 @@
 
 namespace nix {
 
+/**
+ * An enumeration of the ways we can serialize file system objects.
+ */
 enum struct FileIngestionMethod : uint8_t {
+    /**
+     * When we just have a single file, we can directly ingest its
+     * contents
+     */
     Flat = false,
+    /**
+     * For any sort of Nix-support file system object tree, we can
+     * serialize to a ‘Nix Archive’ and then ingest that.
+     */
```

Comment:
The formulation of these comments is a bit weird. Why not make them more descriptive of what these variants do? Something like

```suggestion
    /**
     * Flat-file hashing. Directly ingest the contents of a single file
     */
    Flat = false,
    /**
     * Recursive (or Nar) hashing. Serializes the file-system object in 'NAR' format 
     * and ingest that
     */
```